l10n_it_intrastat_statement/models/intrastat.py

```diff
@@ -0,0 +1,2216 @@
+    invoice_date = fields.Date(string='Invoice Date')
+    intrastat_code_id = fields.Many2one('report.intrastat.code',
+                                        string='Intrastat Code Service')
+    supply_method = fields.Selection([
+        ('I', 'Instant'),
+        ('R', 'Repeatedly'),
+    ], 'Supply Method')
+    payment_method = fields.Selection([
+        ('B', 'Transfer'),
+        ('A', 'Accreditation'),
+        ('X', 'Other'),
+    ], 'Payment Method')
+    country_payment_id = fields.Many2one('res.country', 'Country Payment')
+    invoice_id = fields.Many2one('account.invoice', string='Invoice',
+                                 readonly=True)
+
+    @api.onchange('partner_id')
+    def change_partner_id(self):
+        if self.partner_id:
+            data = self.env['account.invoice.intrastat']._get_partner_data(
+                self.partner_id)
+            self.country_partner_id = data['country_partner_id']
+            self.vat_code = data['vat_code']
+
+    @api.model
+    def _prepare_statement_line(self, inv_intra_line, statement_id):
+        # Period Ref
+        ref_period = statement_id._get_period_ref(inv_intra_line.invoice_id)
+
+        res = {
+            'invoice_id': inv_intra_line.invoice_id.id or False,
+            'month': ref_period and ref_period['month'] or False,
+            'quarterly': ref_period and ref_period['quarterly'] or False,
+            'year_id': ref_period and ref_period['year_id'] or False,
+            'partner_id': inv_intra_line.invoice_id.partner_id.id or False,
+            'country_partner_id':
+                inv_intra_line.country_partner_id.id or False,
+            'vat_code':
+                (
+                    inv_intra_line.invoice_id.partner_id.vat
+                    and inv_intra_line.invoice_id.partner_id.vat[2:]
+                    or False),
+            'amount_euro': self.statement_id.round_min_amount(
+                round(inv_intra_line.amount_euro) or 0,
+                inv_intra_line.invoice_id.company_id),
+            'invoice_number': inv_intra_line.invoice_number or False,
+            'invoice_date': inv_intra_line.invoice_date or False,
+            'intrastat_code_id': inv_intra_line.intrastat_code_id.id or False,
+            'supply_method': inv_intra_line.supply_method or False,
+            'payment_method': inv_intra_line.payment_method or False,
+            'country_payment_id':
+                inv_intra_line.country_payment_id.id or False,
+            'custom_id': statement_id.custom_id.id
+        }
+        return res
+
+    @api.model
+    def _prepare_export_line(self):
+        # Controls
+        # .. Vat code
+        if not self.vat_code:
+            raise ValidationError(
+                _('Missing Vat code for %s in Sale Section 4')
+                % (self.partner_id.name,))
+        # .. year ref
+        if not self.year_id:
+            raise ValidationError(
+                _('Missing Year Ref on 
```

Comment:
Le altre sezioni usano `fields.Date` per `invoice_date`.
C'è un motivo particolare qui per usare `Char`?

Reply:
@primes2h non ho capito cosa intendi : self.invoice_date è usato in tutti i punti del file allo stesso modo.

Reply:
> @primes2h non ho capito cosa intendi : self.invoice_date è usato in tutti i punti del file allo stesso modo.

Intendo dire che nella sezione 3 e 4 di Sale e nella sezione 3 di Purchase `invoice_date` è dichiarato come campo di tipo `Date` mentre nella sezione 4 di Purchase è dichiarato come campo di tipo `Char`.

Reply:
Non l'avevo notato, grazie.
Ho controllato, si può cambiare il campo da Char a Date

Reply:
> Non l'avevo notato, grazie.
> Ho controllato, si può cambiare il campo da Char a Date

Fatto in https://github.com/linkitspa/l10n-italy/pull/12/commits/b12e8ec3fc974edba2e9fdf0a06a3e82f07f278a